test/jdk/tools/jpackage/share/BasicTest.java

```diff
@@ -312,6 +313,14 @@ public void testNoOutputDir(boolean appImage) throws Throwable {
     @Parameter("java.desktop,jdk.jartool")
     @Parameter({ "java.desktop", "jdk.jartool" })
     public void testAddModules(String... addModulesArg) {
+        if (addModulesArg.length == 1 && addModulesArg[0].equals("ALL-MODULE-PATH")) {
+            Path jmods = Path.of(System.getProperty("java.home"), "jmods");
+            boolean noJmods = Files.notExists(jmods);
+            if (LinkableRuntimeImage.isLinkableRuntime() && noJmods) {
+               System.out.println("ALL-MODULE-PATH test skipped for linkable run-time image");
+               return;
+            }
+        }
```

Comment:
The better alternative would be to use `@ParameterSupplier` annotation instead of the list of `@Parameter` to dynamically configure the input for testAddModules() test case. You can use https://github.com/openjdk/jdk/blob/master/test/jdk/tools/jpackage/share/ErrorTest.java or https://github.com/openjdk/jdk/blob/cc628a133e471e7edf07831ff386f0eaf57e9bff/test/jdk/tools/jpackage/helpers-test/jdk/jpackage/test/AnnotationsTest.java#L97 as an example.

Reply:
OK, thank you.

Reply:
Fixed in the update.